Formula/c/chatgpt-cli.rb

```diff
@@ -0,0 +1,31 @@
+class ChatgptCli < Formula
+  desc "CLI for interacting with the OpenAI ChatGPT API"
+  homepage "https://github.com/kardolus/chatgpt-cli"
+  url "https://github.com/kardolus/chatgpt-cli/archive/refs/tags/v1.5.5.tar.gz"
+  sha256 "158620e871411053310c6882a3e164e013389b0b6d99dd6fb42d8d82c988e51d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.GitCommit=homebrew
+      -X main.GitVersion=v#{version}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags.join(" "), output: bin/"chatgpt"), "./cmd/chatgpt"
+    generate_completions_from_executable(bin/"chatgpt", "--set-completions", base_name: "chatgpt")
+  end
+
+  test do
+    config_output = shell_output("#{bin}/chatgpt --config")
+    config_keys = %w[
+      name api_key model max_tokens context_window role temperature
+      top_p frequency_penalty presence_penalty thread omit_history
+      url completions_path models_path auth_header auth_token_prefix
+    ]
+    config_keys.each do |key|
+      assert_match key, config_output
+    end
+  end
```

Comment:
```suggestion
  test do
    assert_match "name: openai", shell_output("#{bin}/chatgpt --config")
    assert_match version.to_s, shell_output("#{bin}/chatgpt --version")
  end
```

@chenrui333's suggestion still applies here. The current test looks like it is more likely prone to break between versions. While the suggestion here (or something similar), ensures that the tool has been installed and has some functionality. 

As previously discussed, note that the purpose of this test is specifically for running `brew test chatgpt-cli` to ensure a functional build, and does not replace other test suites you may have.

Reply:
I see, yeah I never got an answer to my questions about when these tests run. If they run on an upgrade and against the users environment, ie. not against an environment with its own home directory, then this proposed test will actually be flaky and fail for some users. I can propose an update that does not include actual values.

Reply:
Yeah you did: https://github.com/Homebrew/homebrew-core/pull/169625#discussion_r1573682233

Reply:
It also runs in a sandbox with it's own home directory, and mainly on CI machines that will be recreated after every run.